diag_integral/include/diag_integral.inc

```diff
@@ -526,11 +101,11 @@ integer, optional, intent(in) :: js !< starting j indices over which summation i
       field_count (field) = field_count(field) +   &
                             size(data,1)*size(data,2)
       field_sum   (field) = field_sum   (field) +  &
-                            sum (data * area(i1:i2,j1:j2))
+                            sum (real(data,r8_kind) * area(i1:i2,j1:j2))
```

Comment:
real(area(i1:i2,j1:j2),r8_kind?)))

Reply:
this is for my own knowledge, but how come you declare `data` as `FMS_DI_KIND` but then wrap with with `r8_kind` here